extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -419,21 +419,33 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 
 	async findAndExpandParentNode(notebookPath: string): Promise<BookTreeItem> {
 		let bookItem: BookTreeItem = this.currentBook?.getNotebook(notebookPath);
-		// if the node is not expanded getNotebook returns undefined, try to expand the parent node or getChildren of
-		// the root node.
-		if (!bookItem) {
-			// get the parent node and expand it if it's not already
-			let allNodes = this.currentBook?.getAllNotebooks();
-			let book = allNodes ? Array.from(allNodes?.keys())?.filter(x => x.indexOf(notebookPath.substring(0, notebookPath.lastIndexOf(path.sep))) > -1) : undefined;
-			let bookNode = book?.length > 0 ? this.currentBook?.getNotebook(book.find(x => x.substring(0, x.lastIndexOf(path.sep)) === notebookPath.substring(0, notebookPath.lastIndexOf(path.sep)))) : undefined;
-			if (bookNode) {
-				if (this._bookViewer?.visible) {
-					await this._bookViewer.reveal(bookNode, { select: true, focus: false, expand: 3 });
-				} else {
-					await this.getChildren(bookNode);
+		this.currentBook = this.books.length > 1 ? this.books.find(b => notebookPath.indexOf(b.bookPath) > -1) : this.currentBook;
```

Comment:
how can user have two editors active simultaneously, that's the only way two calls can come in at the same time.

Reply:
The calls are asynchronous so if a user opens 2 notebooks quickly after one another there's nothing stopping the calls from being executed in parallel.

In general why do we even need to keep a global "currentBook"? What does "current" even mean?

Reply:
Usually the last opened book (or last operation performed on a notebook in a book) is referred to as currentBook. We don't need to have a global value and can search for it every-time. Initially while implementing the viewlet, I thought it's easier to have a global value so that we don't need to search for it every time. But with revealInViewlet that doesn’t work at all since users open notebooks from search/command-Pallet they don't essentially belong to currentBook.
I'll work on refactoring this in a separate PR.

Reply:
> But why would we assume that it's in the "current" book (whatever that means)? Why not instead search all open books since it could be in any of them?
> > I'm searching all the open books and then setting the appropriate one as current. refer -> this.books.find(b => notebookPath.indexOf(b.bookPath) > -1)

> But we should also be able to just look at the root paths for each of the open books and see if it's a parent of the Notebook, right? I believe that all items in a book have to be under the root folder so that should be enough.
> > Yes, that's how am setting the current book.

> Can books be nested? If so that makes things much more difficult too...
>> according to Jupyter Book documentation, they didn't mention any standard and didn't talk about nesting. But the Livesite has nested books (in some cases to a level 10)